Fix XAML compiler handling of comma/or style selector.

grokys opened the pull request:
## What does the pull request do?

As described in #2838, the XAML compiler didn't handle the comma/or selector syntax. Fixes that.

- `Selector.Or`'s first parameter isn't a `Selector` - remove that check as I don't think it's needed anyway
- Make sure last selector is added to `XamlIlOrSelectorNode`

## Checklist

- [x] Added unit tests (if possible)?

## Fixed issues

Fixes #2838

grokys committed:
Add failing tests for #2838.

grokys committed:
Fix comma (or) style selector.

- `Selector.Or`'s first parameter isn't a `Selector` - remove that check as I don't think it's needed anyway
- Make sure last selector is added to `XamlIlOrSelectorNode`

Fixes #2838

jmacato commented on tests/Avalonia.Markup.Xaml.UnitTests/Xaml/StyleTests.cs:
```diff
@@ -278,5 +278,67 @@ public void Style_Can_Use_Not_Selector()
+<Window xmlns='https://github.com/avaloniaui'
+             xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'>
+    <Window.Styles>
+        <Style Selector='Border,Canvas,ListBox'>
```
wow.. i didnt know this was a thing already! this removes so much redundant styles in many apps out there.

robloo replied:
<img src="https://user-images.githubusercontent.com/17993847/77015121-82660d80-6939-11ea-8d80-a2c1975d7424.jpg" alt="drawing" width="300"/>

kekekeks commented:
`Or` selector will still not work properly though, since currently it doesn't look for a common type and uses the type of the first selector instead: https://github.com/AvaloniaUI/Avalonia/blob/33d3a66c8621f60c8d6e2fb6bab03e00a253bf86/src/Markup/Avalonia.Markup.Xaml/XamlIl/CompilerExtensions/Transformers/AvaloniaXamlIlSelectorTransformer.cs#L315

It should be OK for well-formed XAML

grokys committed:
Merge branch 'master' into fixes/2838-style-or

grokys committed:
Implement finding common base type.

Had to also change the unit test to use controls that all have the `Background` property in a common base type.

grokys commented:
@kekekeks addressed finding the common base type.

kekekeks reviewed (approved):
Awesome

grokys committed:
Merge branch 'master' into fixes/2838-style-or

grokys committed:
Merge branch 'master' into fixes/2838-style-or
